### PR TITLE
make deps: handle dependencies of vendored packages

### PR DESCRIPTION
**Pull request by g2p**
Also drop vendored stuff that was only useful as a dependency,
and forks that are now fixed upstream (ocaml-yaml).

Preparation for PR #47.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,7 +5,8 @@ build:
 
 deps:
 	git submodule update --init --recursive
-	opam install --deps-only -y .
+	opam install --deps-only -y . vendor/*/
+	dune external-lib-deps --missing @@default
```
Comment (pascutto) [inline Makefile]: There should be no missing deps after we call `make deps`, I'm not sure we need this last line.
Also, this new `make deps` does not behave the same way the CI does

**Review by pascutto — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (g2p) [reply Makefile]: The added line doesn't install new deps, it's just a diagnosis to help solve dependencies missing in OPAM.
I'll take a stab at having the CI do the same thing.

Comment (pascutto) [reply Makefile]: I understood it's just a check, I'm saying I'm not sure this should be part of the normal installation workflow for the user (as described in `README.md`), as the OPAM files should always be correct in the `master` branch.

Comment (g2p) [reply Makefile]: Last line removed for now, since the CI wouldn't be calling this code path anymore.

**Comment by pascutto** [comment]
Can't we just add the vendored dependencies to travis PINS list ?

**Comment by g2p** [comment]
Tempting, but then it would be a different kind of pain to keep it updated.
(currently standing at 23 items, obviously generated, and I think that's the lowest it's been since the last mirage retreat)

**Comment by g2p** [comment]
And now we're getting into unrelated failures due to the latest mirage release (and I'm back to wishing for lock files), so I think this change itself has finally matured.

**Comment by g2p** [comment]
(rebased)

**Comment by pascutto** [comment]
Why would we have 23 items when pinning ? Adding pins like `mirage-block:vendor/mirage-block` wouldn't work ?

**Comment by g2p** [comment]
It would work, the point is that it has to be generated from all the opam files.

**Review by pascutto — DISMISSED** [review]

**Review by pascutto — APPROVED** [review]

**Pull request merged**